### PR TITLE
[UBS] - change save employee and update employee #4928

### PR DESCRIPTION
**Pull request by NKorzhik**
# GreenCityUBS PR
https://github.com/ita-social-projects/GreenCity/issues/4928

## Summary Of Changes :fire:
1. Changed SwaggerExampleModel for employee.
2. Change Managemenr Employee Controller for creating, getting and upgrade employee. 
3. Delete unnecessary part of code on ModelUtils.
4. Added tests for code to ManagementEmployeeControllerTest. 
5. Added new query to DB in EmployeeRepository and TariffsInfoRepository.
6. Delete table employee-receiving-station-mapping from DB.
7. Add new lines of code to UserRemoteClient and  UserRemoteClientFallbackFactory.
8. Added new Dto for correct working of new functional.
9. Delete unnecessary Dtos.
10. Change UBSManagementEmployeeServiceImpl.
11. Added tests to UBSManagementEmployeeServiceImplTest.

## Added
* ch-drop-table-employee-receiving-station-mapping-Korzh.xml
* GetReceivingStationDto.java
* GetEmployeeDto
* UpdateEmployeeAuthoritiesDto
* GetTariffInfoForEmployeeDto
* EmployeeUpdateDtoMapper
* GetEmployeeDtoMapper
* UpdateEmployeeDtoMapper
* GetTariffInfoForEmployeeDtoMapper
* GetEmployeeDtoMapperTest
* UpdateEmployeeDtoMapperTest
* GetTariffInfoForEmployeeDtoMapperTest

## Changed
* SwaggerExampleModel
* ManagementEmployeeController.java
* ModelUtils.java
* ManagementEmployeeControllerTest.java
* EmployeeRepository.java
* TariffsInfoRepository.java
* db.changelog-master.xml
* UserRemoteClient.java
* UserRemoteClientFallbackFactory.java
* ErrorMessage.java
* RegionDto.java
* EmployeeDto.java
* EmployeeSignUpDto.java
* UBSManagementEmployeeService.java
* UserRemoteClientFallbackFactoryTest.java
* UBSManagementEmployeeServiceImpl.java
* ModelUtils.java
* UBSManagementEmployeeServiceImplTest.java

## Deleted
* EmployeeMapper.java
* EmployeeDtoMapper.java
* EmployeeDtoMapperTest.java

# PR Checklist Forms

- [x]  СI passed
- [x]  Сode coverage >=95%
- [x]  PR is reviewed manually again (to make sure you have 100% ready code)
- [x]  All reviewers agreed to merge the PR
- [x]  I've checked new feature as logged in and logged out user if needed
- [x]  PR meets all conventions

### CODE REVIEW

File: core/src/main/java/greencity/controller/ManagementEmployeeController.java [hunk 1]
```diff
@@ -288,4 +289,21 @@ public ResponseEntity<Object> editAuthorities(@Valid @RequestBody UserEmployeeAu
...
+     * @return list of all tariffs.
+     */
+    @ApiOperation(value = "Get all tariffs for working with employee page")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = HttpStatuses.OK),
+        @ApiResponse(code = 401, message = HttpStatuses.UNAUTHORIZED),
+        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN)
+    })
+    @PreAuthorize("@preAuthorizer.hasAuthority('SEE_EMPLOYEES_PAGE', authentication)")
+    @GetMapping("/getTariffs")
```
Comment (VadymMakitra) [inline core/src/main/java/greencity/controller/ManagementEmployeeController.java:306]: Please follow the name convention of API according to the REST standard.

File: core/src/test/java/greencity/controller/ManagementEmployeeControllerTest.java [hunk 2]
```diff
@@ -77,20 +77,20 @@ void setup() {
...
+//        String responseJSON = objectMapper.writeValueAsString(dto);
+//        MockMultipartFile jsonFile = new MockMultipartFile("employeeDto",
+//            "", "application/json", responseJSON.getBytes());
+//
+//        mockMvc.perform(multipart(UBS_LINK + SAVE_LINK)
+//            .file(jsonFile)
+//            .principal(principal)
+//            .contentType(MediaType.APPLICATION_JSON))
+//            .andExpect(status().isCreated());
+//    }
```
Comment (VadymMakitra) [inline core/src/test/java/greencity/controller/ManagementEmployeeControllerTest.java]: What is the reason for commenting on this test?

File: core/src/test/java/greencity/controller/ManagementEmployeeControllerTest.java [hunk 3]
```diff
@@ -124,25 +124,25 @@ void getAllActiveEmployees() throws Exception {
...
+//        builder.with(request -> {
+//            request.setMethod("PUT");
+//            return request;
+//        });
+//
+//        mockMvc.perform(builder.file(jsonFile)
+//            .principal(principal)
+//            .contentType(MediaType.APPLICATION_JSON))
+//            .andExpect(status().isBadRequest());
+//    }
```
Comment (VadymMakitra) [inline core/src/test/java/greencity/controller/ManagementEmployeeControllerTest.java]: What is the reason for commenting on this test?

File: dao/src/main/java/greencity/repository/EmployeeRepository.java [hunk 4]
```diff
@@ -64,6 +64,17 @@ public interface EmployeeRepository extends JpaRepository<Employee, Long> {
...
+    /**
+     * Method that return list of Employee with email except with current id.
+     *
+     * @param email {@link String}
+     * @param id    {@link Long}
+     * @return list of employee;
+     * @author Nikita Korzh
+     */
+    @Query(value = "SELECT em from Employee em where em.email=:email and not em.id =:id")
+    List<Employee> findEmployeeWithEmailAndNotId(String email, Long id);
```
Comment (VadymMakitra) [inline dao/src/main/java/greencity/repository/EmployeeRepository.java]: Spring JPA could provide this query.

File: dao/src/main/java/greencity/repository/TariffsInfoRepository.java [hunk 5]
```diff
@@ -43,4 +45,14 @@ Optional<TariffsInfo> findTariffsInfoLimitsByCourierIdAndLocationId(@Param("cour
...
+
+    /**
+     * Method for getting set of tariffs.
+     *
+     * @param tariffId - list of tariffIds.
+     * @return - set of tariffs.
+     * @author - Nikita Korzh.
+     */
+    @Query("SELECT ti FROM TariffsInfo ti where ti.id in(:tariffId)")
+    Set<TariffsInfo> findTariffsInfoById(@Param("tariffId") List<Long> tariffId);
```
Comment (VadymMakitra) [inline dao/src/main/java/greencity/repository/TariffsInfoRepository.java]: Spring JPA could provide this query.

File: service-api/src/main/java/greencity/client/UserRemoteClient.java [hunk 6]
```diff
@@ -111,6 +112,14 @@ public interface UserRemoteClient {
      * @param employeeEmail    {@link String}
      * @param newEmployeeEmail {@link String}
      */
-    @PutMapping("user/update-employee-email")
+    @PutMapping("/user/update-employee-email")
```
Comment (VadymMakitra) [inline service-api/src/main/java/greencity/client/UserRemoteClient.java]: Please follow the name convention of API according to the REST standard.

File: service-api/src/main/java/greencity/client/UserRemoteClient.java [hunk 7]
```diff
@@ -111,6 +112,14 @@ public interface UserRemoteClient {
...
     void updateEmployeeEmail(@RequestParam String employeeEmail, @RequestParam String newEmployeeEmail);
+
+    /**
+     * Update authorities in chosen employee.
+     *
+     * @param dto contains email and list of positions.
+     */
+    @PutMapping("/user/update-authorities")
```
Comment (VadymMakitra) [inline service-api/src/main/java/greencity/client/UserRemoteClient.java]: Please follow the name convention of API according to the REST standard.

File: service-api/src/main/java/greencity/dto/courier/GetReceivingStationDto.java [hunk 8]
```diff
@@ -0,0 +1,15 @@
+package greencity.dto.courier;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Builder
```
Comment (VadymMakitra) [inline service-api/src/main/java/greencity/dto/courier/GetReceivingStationDto.java]: Is it required to have setters and builders?

File: service-api/src/main/java/greencity/dto/RegionDto.java [hunk 9]
```diff
@@ -1,14 +1,14 @@
...
-import lombok.*;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
 
-@Getter
-@Setter
+@Data
+@Builder
```
Comment (VadymMakitra) [inline service-api/src/main/java/greencity/dto/RegionDto.java]: Is it required to have setters and builders?

File: service-api/src/main/java/greencity/dto/employee/GetEmployeeDto.java [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import javax.validation.constraints.*;
+import java.util.List;
+import java.util.Set;
+
+@Data
+@Builder
```
Comment (VadymMakitra) [inline service-api/src/main/java/greencity/dto/employee/GetEmployeeDto.java]: Is it required to have setters and builders?

File: service-api/src/main/java/greencity/dto/employee/SaveEmployeeDto.java [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+import lombok.NoArgsConstructor;
+
+import javax.validation.constraints.Email;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+import java.util.List;
+
+@Data
+@Builder
```
Comment (VadymMakitra) [inline service-api/src/main/java/greencity/dto/employee/SaveEmployeeDto.java]: Is it required to have setters and builders?

File: service-api/src/main/java/greencity/dto/employee/UpdateEmployeeAuthoritiesDto.java [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+import greencity.dto.position.PositionDto;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.List;
+
+@Data
+@Builder
```
Comment (VadymMakitra) [inline service-api/src/main/java/greencity/dto/employee/UpdateEmployeeAuthoritiesDto.java]: Is it required to have setters and builders?

File: service-api/src/main/java/greencity/dto/tariff/GetTariffInfoForEmployeeDto.java [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.List;
+import java.util.Optional;
+
+@Data
+@Builder
```
Comment (VadymMakitra) [inline service-api/src/main/java/greencity/dto/tariff/GetTariffInfoForEmployeeDto.java]: Is it required to have setters and builders?

File: dao/src/main/java/greencity/repository/EmployeeRepository.java (hunk 4 above)
Comment (NKorzhik) [reply dao/src/main/java/greencity/repository/EmployeeRepository.java]: done

File: dao/src/main/java/greencity/repository/TariffsInfoRepository.java (hunk 5 above)
Comment (NKorzhik) [reply dao/src/main/java/greencity/repository/TariffsInfoRepository.java]: done

File: service-api/src/main/java/greencity/dto/employee/SaveEmployeeDto.java (hunk 11 above)
Comment (NKorzhik) [reply service-api/src/main/java/greencity/dto/employee/SaveEmployeeDto.java]: done

File: service-api/src/main/java/greencity/dto/employee/GetEmployeeDto.java (hunk 10 above)
Comment (NKorzhik) [reply service-api/src/main/java/greencity/dto/employee/GetEmployeeDto.java]: done

File: service-api/src/main/java/greencity/dto/employee/UpdateEmployeeAuthoritiesDto.java (hunk 12 above)
Comment (NKorzhik) [reply service-api/src/main/java/greencity/dto/employee/UpdateEmployeeAuthoritiesDto.java]: done

File: service-api/src/main/java/greencity/dto/tariff/GetTariffInfoForEmployeeDto.java (hunk 13 above)
Comment (NKorzhik) [reply service-api/src/main/java/greencity/dto/tariff/GetTariffInfoForEmployeeDto.java]: done

File: service-api/src/main/java/greencity/dto/RegionDto.java (hunk 9 above)
Comment (NKorzhik) [reply service-api/src/main/java/greencity/dto/RegionDto.java]: done

File: core/src/main/java/greencity/controller/ManagementEmployeeController.java (hunk 1 above)
Comment (NKorzhik) [reply core/src/main/java/greencity/controller/ManagementEmployeeController.java:306]: I think, that I should use /tariffs, but this one is already in use. So what I need to do?

File: service-api/src/main/java/greencity/client/UserRemoteClient.java (hunk 7 above)
Comment (NKorzhik) [reply service-api/src/main/java/greencity/client/UserRemoteClient.java]: done, i deleted the update from url, because I hane not to show which crud operation I use

File: service-api/src/main/java/greencity/client/UserRemoteClient.java (hunk 6 above)
Comment (NKorzhik) [reply service-api/src/main/java/greencity/client/UserRemoteClient.java]: done, i deleted the update from url, because I hane not to show which crud operation I use

Comment (NKorzhik) [reply service-api/src/main/java/greencity/client/UserRemoteClient.java]: done, i deleted the update from url, because I hane not to show which crud operation I use

File: core/src/main/java/greencity/controller/ManagementEmployeeController.java [hunk 14]
```diff
@@ -3,11 +3,11 @@
 import greencity.annotations.ApiPageable;
 import greencity.constants.HttpStatuses;
 import greencity.constants.SwaggerExampleModel;
-import greencity.dto.employee.EmployeeDto;
-import greencity.dto.employee.UserEmployeeAuthorityDto;
+import greencity.dto.employee.*;
```
Comment (RomaMocherniuk) [inline core/src/main/java/greencity/controller/ManagementEmployeeController.java]: please do not use `*`

File: service-api/src/main/java/greencity/client/UserRemoteClient.java [hunk 15]
```diff
@@ -106,11 +107,19 @@ public interface UserRemoteClient {
...
      */
-    @PutMapping("user/update-employee-email")
-    void updateEmployeeEmail(@RequestParam String employeeEmail, @RequestParam String newEmployeeEmail);
+    @PutMapping("/user/employee-email")
+    void updateEmployeeEmail(@RequestParam String newEmployeeEmail, @RequestParam String uuid);
+
+    /**
+     * Update authorities in chosen employee.
+     *
+     * @param dto contains email and list of positions.
```
Comment (RomaMocherniuk) [inline service-api/src/main/java/greencity/client/UserRemoteClient.java]: please add a link to the `UpdateEmployeeAuthoritiesDto`

File: service-api/src/main/java/greencity/client/config/UserRemoteClientFallbackFactory.java [hunk 16]
```diff
@@ -67,17 +69,24 @@ public void updateEmployeesAuthorities(UserEmployeeAuthorityDto dto, String emai
                 throw new RemoteServerUnavailableException(ErrorMessage.EMPLOYEE_AUTHORITY_WAS_NOT_EDITED, throwable);
             }
 
+            @PostMapping("/ownSecurity/sign-up-employee")
```
Comment (RomaMocherniuk) [inline service-api/src/main/java/greencity/client/config/UserRemoteClientFallbackFactory.java]: do we need this mapping here?

File: service-api/src/main/java/greencity/constant/ErrorMessage.java [hunk 17]
```diff
@@ -119,6 +120,8 @@ public final class ErrorMessage {
         "Limits are not inputted properly. You should only input AmountOfBags or PriceOfOrder.";
     public static final String MAX_BAG_VALUE_IS_INCORRECT = "Max bag value should be greater than min";
     public static final String MAX_PRICE_VALUE_IS_INCORRECT = "Max price value should be greater than min";
+    public static final String USER_NOT_FOUND = "User with current email doesn't exist: ";
+    public static final String USER_NOT_FOUND_BY_PHONE = "User with current phone number doesn't exist: ";
```
Comment (RomaMocherniuk) [inline service-api/src/main/java/greencity/constant/ErrorMessage.java]: never used

File: service-api/src/main/java/greencity/constant/ErrorMessage.java [hunk 18]
```diff
@@ -119,6 +120,8 @@ public final class ErrorMessage {
         "Limits are not inputted properly. You should only input AmountOfBags or PriceOfOrder.";
     public static final String MAX_BAG_VALUE_IS_INCORRECT = "Max bag value should be greater than min";
     public static final String MAX_PRICE_VALUE_IS_INCORRECT = "Max price value should be greater than min";
+    public static final String USER_NOT_FOUND = "User with current email doesn't exist: ";
```
Comment (RomaMocherniuk) [inline service-api/src/main/java/greencity/constant/ErrorMessage.java]: never used

File: service-api/src/main/java/greencity/dto/courier/GetReceivingStationDto.java [hunk 19]
```diff
@@ -0,0 +1,14 @@
+package greencity.dto.courier;
+
+import lombok.*;
```
Comment (RomaMocherniuk) [inline service-api/src/main/java/greencity/dto/courier/GetReceivingStationDto.java]: please do not use `*`

File: service-api/src/main/java/greencity/dto/employee/EmployeeDto.java [hunk 20]
```diff
@@ -1,27 +1,20 @@
...
 import javax.validation.constraints.*;
 import java.util.List;
-import java.util.Set;
 
 @Data
 @Builder
 @NoArgsConstructor
 @AllArgsConstructor
 public class EmployeeDto {
-    @Min(1)
```
Comment (RomaMocherniuk) [inline service-api/src/main/java/greencity/dto/employee/EmployeeDto.java]: any reason you removed it?

File: service/src/main/java/greencity/mapping/employee/UpdateEmployeeDtoMapper.java [hunk 21]
```diff
@@ -0,0 +1,31 @@
...
+public class UpdateEmployeeDtoMapper extends AbstractConverter<EmployeeDto, Employee> {
+    @Override
+    protected Employee convert(EmployeeDto employeeDto) {
+        return Employee.builder()
+            .id(employeeDto.getId())
+            .firstName(employeeDto.getFirstName())
+            .lastName(employeeDto.getLastName())
+            .email(employeeDto.getEmail())
+            .phoneNumber(employeeDto.getPhoneNumber())
+            .employeeStatus(EmployeeStatus.ACTIVE)
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/mapping/employee/UpdateEmployeeDtoMapper.java]: should the status be ACTIVE every time?

File: service/src/main/java/greencity/mapping/tariff/GetTariffInfoForEmployeeDtoMapper.java [hunk 22]
```diff
@@ -0,0 +1,53 @@
...
+
+@Component
+public class GetTariffInfoForEmployeeDtoMapper extends AbstractConverter<TariffsInfo, GetTariffInfoForEmployeeDto> {
+    @Override
+    protected GetTariffInfoForEmployeeDto convert(TariffsInfo source) {
+        Region region = source.getTariffLocations() != null
+            ? source.getTariffLocations().iterator().next().getLocation().getRegion()
+            : null;
+        RegionDto regionDto = region != null ? RegionDto.builder().regionId(region.getId()).nameEn(region.getEnName())
+            .nameUk(region.getUkrName()).build() : null;
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/mapping/tariff/GetTariffInfoForEmployeeDtoMapper.java]: please move it to the method

File: service/src/main/java/greencity/mapping/tariff/GetTariffInfoForEmployeeDtoMapper.java [hunk 23]
```diff
@@ -0,0 +1,53 @@
...
+
+        List<LocationsDtos> locationsDtos = source.getTariffLocations().stream()
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/mapping/tariff/GetTariffInfoForEmployeeDtoMapper.java:27]: I see you check if `source.getTariffLocations()` https://github.com/ita-social-projects/GreenCityUBS/pull/955/files#diff-48d54bbc1999db3f0a3864df1665781366165526155d4d9d706ab791583c46dcR20
It means that we can have `NPE` here

File: service/src/main/java/greencity/service/ubs/UBSManagementEmployeeServiceImpl.java [hunk 24]
```diff
@@ -1,18 +1,18 @@
 package greencity.service.ubs;
 
+import com.netflix.hystrix.exception.HystrixRuntimeException;
 import greencity.client.UserRemoteClient;
 import greencity.constant.AppConstant;
 import greencity.constant.ErrorMessage;
-import greencity.dto.courier.ReceivingStationDto;
-import greencity.dto.employee.EmployeeDto;
-import greencity.dto.employee.EmployeeSignUpDto;
+import greencity.dto.employee.*;
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/ubs/UBSManagementEmployeeServiceImpl.java]: please do not use *

File: service/src/test/java/greencity/ModelUtils.java [hunk 25]
```diff
@@ -7,12 +7,7 @@
 import com.google.maps.model.Geometry;
 import com.google.maps.model.LatLng;
 import greencity.constant.AppConstant;
-import greencity.dto.AddNewTariffDto;
-import greencity.dto.CreateAddressRequestDto;
-import greencity.dto.DetailsOfDeactivateTariffsDto;
-import greencity.dto.LocationsDtos;
-import greencity.dto.OptionForColumnDTO;
-import greencity.dto.TariffsForLocationDto;
+import greencity.dto.*;
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/ModelUtils.java]: please do not use *

File: service/src/test/java/greencity/ModelUtils.java [hunk 26]
```diff
@@ -27,22 +22,10 @@
 import greencity.dto.certificate.CertificateDto;
 import greencity.dto.certificate.CertificateDtoForAdding;
 import greencity.dto.certificate.CertificateDtoForSearching;
-import greencity.dto.courier.CourierDto;
-import greencity.dto.courier.CourierTranslationDto;
-import greencity.dto.courier.CourierUpdateDto;
-import greencity.dto.courier.CreateCourierDto;
-import greencity.dto.courier.ReceivingStationDto;
+import greencity.dto.courier.*;
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/ModelUtils.java]: same here

File: service/src/test/java/greencity/ModelUtils.java [hunk 27]
```diff
@@ -27,22 +22,10 @@
...
-import greencity.dto.employee.AddEmployeeDto;
-import greencity.dto.employee.EmployeeDto;
-import greencity.dto.employee.EmployeeNameDto;
-import greencity.dto.employee.EmployeeNameIdDto;
-import greencity.dto.employee.EmployeePositionDtoRequest;
-import greencity.dto.employee.EmployeePositionDtoResponse;
-import greencity.dto.employee.EmployeeSignUpDto;
-import greencity.dto.employee.UpdateResponsibleEmployeeDto;
-import greencity.dto.employee.UserEmployeeAuthorityDto;
+import greencity.dto.employee.*;
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/ModelUtils.java]: and here

File: service/src/test/java/greencity/ModelUtils.java [hunk 28]
```diff
@@ -105,11 +87,7 @@
 import greencity.dto.service.AddServiceDto;
 import greencity.dto.service.CreateServiceDto;
 import greencity.dto.service.ServiceDto;
-import greencity.dto.tariff.EditTariffServiceDto;
-import greencity.dto.tariff.GetTariffsInfoDto;
-import greencity.dto.tariff.SetTariffLimitsDto;
-import greencity.dto.tariff.TariffTranslationDto;
-import greencity.dto.tariff.TariffsInfoDto;
+import greencity.dto.tariff.*;
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/ModelUtils.java]: and here

**Comment by RomaMocherniuk** [comment]
It is a very big PR.
Please test it very carefully

File: service/src/main/java/greencity/mapping/tariff/GetTariffInfoForEmployeeDtoMapper.java [hunk 29]
```diff
@@ -0,0 +1,53 @@
...
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class GetTariffInfoForEmployeeDtoMapper extends AbstractConverter<TariffsInfo, GetTariffInfoForEmployeeDto> {
+    @Override
+    protected GetTariffInfoForEmployeeDto convert(TariffsInfo source) {
+        Region region = source.getTariffLocations() != null
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/mapping/tariff/GetTariffInfoForEmployeeDtoMapper.java]: please add test if `source.getTariffLocations() != null`

File: service/src/main/java/greencity/mapping/tariff/GetTariffInfoForEmployeeDtoMapper.java [hunk 30]
```diff
@@ -0,0 +1,53 @@
...
+            ? source.getTariffLocations().iterator().next().getLocation().getRegion()
+            : null;
+        RegionDto regionDto = region != null ? RegionDto.builder().regionId(region.getId()).nameEn(region.getEnName())
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/mapping/tariff/GetTariffInfoForEmployeeDtoMapper.java]: please add test if `region != null`

File: service/src/main/java/greencity/service/ubs/UBSManagementEmployeeServiceImpl.java [hunk 31]
```diff
@@ -224,41 +235,40 @@ public void deletePosition(Long id) {
...
+    private void updateEmployeeEmail(EmployeeDto dto, String uuid) {
         Employee employee = employeeRepository.findById(dto.getId())
             .orElseThrow(() -> new NotFoundException(ErrorMessage.EMPLOYEE_NOT_FOUND + dto.getId()));
         String oldEmail = employee.getEmail();
         String newEmail = dto.getEmail();
-        userRemoteClient.updateEmployeeEmail(oldEmail, newEmail);
+        if (!oldEmail.equals(newEmail)) {
+            try {
+                userRemoteClient.updateEmployeeEmail(newEmail, uuid);
+            } catch (HystrixRuntimeException e) {
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/ubs/UBSManagementEmployeeServiceImpl.java:260]: please add test for this case

File: core/src/main/java/greencity/controller/ManagementEmployeeController.java (hunk 14 above)
Comment (ABbondar) [reply core/src/main/java/greencity/controller/ManagementEmployeeController.java]: fixed

File: service-api/src/main/java/greencity/client/UserRemoteClient.java (hunk 15 above)
Comment (ABbondar) [reply service-api/src/main/java/greencity/client/UserRemoteClient.java]: added

File: service-api/src/main/java/greencity/dto/courier/GetReceivingStationDto.java (hunk 19 above)
Comment (ABbondar) [reply service-api/src/main/java/greencity/dto/courier/GetReceivingStationDto.java]: fixed

File: service/src/main/java/greencity/service/ubs/UBSManagementEmployeeServiceImpl.java (hunk 24 above)
Comment (ABbondar) [reply service/src/main/java/greencity/service/ubs/UBSManagementEmployeeServiceImpl.java]: fixed

File: service/src/test/java/greencity/ModelUtils.java (hunk 26 above)
Comment (ABbondar) [reply service/src/test/java/greencity/ModelUtils.java]: fixed

File: service/src/test/java/greencity/ModelUtils.java (hunk 27 above)
Comment (ABbondar) [reply service/src/test/java/greencity/ModelUtils.java]: fixed

File: service/src/test/java/greencity/ModelUtils.java (hunk 28 above)
Comment (ABbondar) [reply service/src/test/java/greencity/ModelUtils.java]: fixed

File: service/src/main/java/greencity/service/ubs/UBSManagementEmployeeServiceImpl.java (hunk 31 above)
Comment (ABbondar) [reply service/src/main/java/greencity/service/ubs/UBSManagementEmployeeServiceImpl.java:260]: added

File: service/src/main/java/greencity/mapping/tariff/GetTariffInfoForEmployeeDtoMapper.java (hunk 30 above)
Comment (ABbondar) [reply service/src/main/java/greencity/mapping/tariff/GetTariffInfoForEmployeeDtoMapper.java]: added

File: service/src/test/java/greencity/ModelUtils.java (hunk 25 above)
Comment (ABbondar) [reply service/src/test/java/greencity/ModelUtils.java]: fixed

File: service/src/main/java/greencity/mapping/tariff/GetTariffInfoForEmployeeDtoMapper.java (hunk 29 above)
Comment (ABbondar) [reply service/src/main/java/greencity/mapping/tariff/GetTariffInfoForEmployeeDtoMapper.java]: added

File: service-api/src/main/java/greencity/client/config/UserRemoteClientFallbackFactory.java (hunk 16 above)
Comment (NKorzhik) [reply service-api/src/main/java/greencity/client/config/UserRemoteClientFallbackFactory.java]: no, i deleted

File: service-api/src/main/java/greencity/constant/ErrorMessage.java (hunk 17 above)
Comment (NKorzhik) [reply service-api/src/main/java/greencity/constant/ErrorMessage.java]: deleted

File: service-api/src/main/java/greencity/constant/ErrorMessage.java (hunk 18 above)
Comment (NKorzhik) [reply service-api/src/main/java/greencity/constant/ErrorMessage.java]: same

File: service/src/main/java/greencity/mapping/employee/UpdateEmployeeDtoMapper.java (hunk 21 above)
Comment (NKorzhik) [reply service/src/main/java/greencity/mapping/employee/UpdateEmployeeDtoMapper.java]: fixed

File: service-api/src/main/java/greencity/dto/employee/EmployeeDto.java (hunk 20 above)
Comment (NKorzhik) [reply service-api/src/main/java/greencity/dto/employee/EmployeeDto.java]: I returned it

File: service/src/main/java/greencity/mapping/tariff/GetTariffInfoForEmployeeDtoMapper.java (hunk 23 above)
Comment (NKorzhik) [reply service/src/main/java/greencity/mapping/tariff/GetTariffInfoForEmployeeDtoMapper.java:27]: fixed

File: service/src/main/java/greencity/mapping/tariff/GetTariffInfoForEmployeeDtoMapper.java (hunk 22 above)
Comment (NKorzhik) [reply service/src/main/java/greencity/mapping/tariff/GetTariffInfoForEmployeeDtoMapper.java]: fixed

File: core/src/test/java/greencity/controller/ManagementEmployeeControllerTest.java (hunk 3 above)
Comment (ABbondar) [reply core/src/test/java/greencity/controller/ManagementEmployeeControllerTest.java]: fixed

File: core/src/test/java/greencity/controller/ManagementEmployeeControllerTest.java (hunk 2 above)
Comment (ABbondar) [reply core/src/test/java/greencity/controller/ManagementEmployeeControllerTest.java]: fixed

File: service-api/src/main/java/greencity/dto/employee/UpdateEmployeeAuthoritiesDto.java [hunk 32]
```diff
@@ -0,0 +1,22 @@
...
+
+@Getter
+@NoArgsConstructor
+@AllArgsConstructor
+@EqualsAndHashCode
+@ToString
+@Builder
+public class UpdateEmployeeAuthoritiesDto {
+    String email;
+    List<PositionDto> positions;
```
Comment (RomaMocherniuk) [inline service-api/src/main/java/greencity/dto/employee/UpdateEmployeeAuthoritiesDto.java]: `private`

**Review by RomaMocherniuk — APPROVED** [review]

**Review by ABbondar — APPROVED** [review]

**Review by OlegVat — APPROVED** [review]

**Review by juseti — APPROVED** [review]

**Review by JJJazl — APPROVED** [review]
